manifests/kiali-ossm/manifests/kiali.crd.yaml

```diff
@@ -21,4 +22,1707 @@ spec:
+                    type: string
+                  image_pull_policy:
+                    description: "The Kubernetes pull policy for the Kiali deployment. This is overridden to be 'Always' if `deployment.image_version` is set to 'latest'."
+                    type: string
+                    default: "IfNotPresent"
+                  image_pull_secrets:
+                    description: "The names of the secrets to be used when container images are to be pulled."
+                    type: array
+                    items:
+                      type: string
+                  image_version:
+                    description: |
+                      Determines which version of Kiali to install.
+                      Choose 'lastrelease' to use the last Kiali release.
+                      Choose 'latest' to use the latest image (which may or may not be a released version of Kiali).
+                      Choose 'operator_version' to use the image whose version is the same as the operator version.
+                      Otherwise, you can set this to any valid Kiali version (such as 'v1.0') or any valid Kiali
+                      digest hash (if you set this to a digest hash, you must indicate the digest in `deployment.image_digest`).
+
+                      Note that if this is set to 'latest' then the `deployment.image_pull_policy` will be set to 'Always'.
+
+                      If you set this to a specific version (i.e. you do not leave it as the default empty string),
+                      you must make sure that image is supported by the operator.
+
+                      If empty, the operator will use a known supported image version based on which 'version' was defined.
+                      Note that, as a security measure, a cluster admin may have configured the Kiali operator to
+                      ignore this setting. A cluster admin may do this to ensure the Kiali operator only installs
+                      a single, specific Kiali version, thus this setting may have no effect depending on how the
+                      operator itself was configured.
+                    type: string
+                  ingress:
+                    description: "Configures if/how the Kiali endpoint should be exposed externally."
+                    type: object
+                    properties:
+                      additional_labels:
+                        description: "Additional labels to add to the Ingress (or Route if on OpenShift). These are added to the labels that are created by default; these do not override the default labels."
+                        type: object
+                        x-kubernetes-preserve-unknown-fields: true
```

Comment:
Why are we preserving unknown fields for nearly all objects in the CRD?

Reply:
I don't think you need this. This is from the kube API itself:
```
          "labels": {
            "description": "Map of string keys and values that can be used to organize and categorize (scope and select) objects. May match selectors of replication controllers and services. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels",
            "type": "object",
            "additionalProperties": {
              "type": "string",
              "default": ""
            }
          },
```

Reply:
My only reason to be nitpicky is that once we publish the schema it will be hard to change it. With `x-kubernetes-preserve-unknown-fields` I think you could have something like
```
additionalLabels:
  0: false
  more:
    unknownfields
```

Reply:
So, `type: object` would be all that is needed? Just remove `x-kubernetes-preserve-unknown-fields: true` .. I will test that out.

Reply:
> So, type: object would be all that is needed? Just remove x-kubernetes-preserve-unknown-fields: true .. I will test that out.

I believe so. At least that's what upstream types look like from the `/openapi/v3` endpoint.